### PR TITLE
Add IonQTargetGateset and use it for circuit compilations.

### LINKED ISSUE #5129: [IONQ] Vendor Transformer adoption
**Summarize the task**
With broader changes to devices (https://github.com/quantumlib/Cirq/issues/4744) and optimizers/trasnformers (https://github.com/quantumlib/Cirq/issues/4722), vendors might need to update device decomposition to use the newest transformer APIs. At first glance it looks like:

-----------------------------------

`cirq-ionq` has their own devices and should be in a good position to consolidate any missing device info into the metadata fields as well as adapt `decompose_to_device` into a proper transformer.

https://github.com/quantumlib/Cirq/blob/45624ff7259f53589c2139c97108d8bf11502173/cirq-ionq/cirq_ionq/ionq_devices.py#L106

@ColemanCollins @nakardo @gmauricio

**Acceptance criteria - when is the task considered done?**

**Related**

Related issues: #4901 , #5127

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes #5129 #4901 #4153
Replaces https://github.com/quantumlib/Cirq/pull/5127

* Adds support for compiling > 2q operations to ionq's target gateset
* Does not do the "merge 2q operations to a connected component" optimization, to preserve existing behavior. We can add another `preprocess_transformer` if we want to do the optimization. 
* Deprecates `cirq_ionq.decompose_to_device`  in favour of using `cirq.optimize_for_target_gateset(circuit, gateset=cirq_ionq.IonQTargetGateset)` for compiling circuits. 

cc @Cynocracy @dabacon

### CODE REVIEW

**Comment by Cynocracy** [comment]
Thanks for picking this up!

File: cirq-ionq/cirq_ionq/ionq_gateset.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# limitations under the License.
+
+"""Target gateset used for compiling circuits to IonQ device."""
+from typing import Any
+from typing import Dict
+from typing import List
+
+import cirq
+from cirq.transformers.target_gatesets.compilation_target_gateset import (
+    _create_transformer_with_kwargs,
```
Comment (dabacon) [inline cirq-ionq/cirq_ionq/ionq_gateset.py:22]: Should we be making this public?  This creates a dependency on into cirq that is "private" and so subject to change.

File: cirq-ionq/cirq_ionq/ionq_gateset.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+)
+
+
+class IonQTargetGateset(cirq.TwoQubitCompilationTargetGateset):
+    """Target gateset for compiling circuits to IonQ devices.
+
+    The gates supported by the API are
+    * `cirq.XPowGate`, `cirq.YPowGate`, `cirq.ZPowGate`
+    * `cirq.XXPowGate`, `cirq.YYPowGate`, `cirq.ZZPowGate`
+    * `cirq.CNOT`, `cirq.H`, `cirq.SWAP`
```
Comment (dabacon) [inline cirq-ionq/cirq_ionq/ionq_gateset.py:32]: nit: maybe move H up to first line, since it is single qubit standard gate.

**Review by dabacon — APPROVED** [review]
Two small comments, but LGTM

File: cirq-ionq/cirq_ionq/ionq_gateset.py (hunk 2 above)
Comment (tanujkhattar) [reply cirq-ionq/cirq_ionq/ionq_gateset.py]: Updated the docstring.

File: cirq-ionq/cirq_ionq/ionq_gateset.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-ionq/cirq_ionq/ionq_gateset.py:22]: agreed, filed https://github.com/quantumlib/Cirq/issues/5491 to track this change.

**Pull request merged**